### PR TITLE
fix: Added hook to combobox for when options change

### LINKED ISSUE #2182: [fix] when ComboBox options prop is updated displayed options do not change
**ReactUSWDS Version & USWDS Version:**

react-uswds: 3.1.0
uswds: 2.13.3

**Describe the bug**

When ComboBox `options` prop is updated displayed options do not change

gif demonstrating issue

Seems like the component should watch for changes to the prop and dispatch an update options action

**To Reproduce**

See: https://github.com/yeonsy/bug-report/tree/main/react-uswds/combobox-options-update#readme

**Expected behavior**

Displayed options should be synced with `options` prop

**Device and Browser Information (please complete the following information if describing a UI bug):**

Firefox 102

### PR DESCRIPTION
**Pull request by werdnanoslen**
# Summary

- added hook to combobox for when options change
- added test
- updated story to use this feature
- added veggies to food list and renamed file

## Related Issues or PRs

Resolves #2182

## How To Test

- Run new test
- See updated story

### Screenshots (optional)
Jan-19-2024 10-59-17

### CODE REVIEW

**Comment by werdnanoslen** [comment]
This may cause new behaviors, so might be a breaking change?

File: src/components/forms/ComboBox/ComboBox.tsx
Comment (brandonlenz) [reply src/components/forms/ComboBox/ComboBox.tsx:151]: I suspect this useEffect should dispatch an event that both changes the options to filter from, _and_ filters the list if the user has already typed into the combo box when the options change.

Comment (werdnanoslen) [reply src/components/forms/ComboBox/ComboBox.tsx:151]: I don't think so... e.g. I added a little 'test' console log right above the dispatch:
https://github.com/trussworks/react-uswds/blob/9ac7a45d7cc5e330ee7ed068e4923e62520120b9/src/components/forms/ComboBox/ComboBox.tsx#L399-L400

And it doesn't seem to fire when the options change, just when the input is typed in:
Feb-07-2024 10-07-17

Comment (brandonlenz) [reply src/components/forms/ComboBox/ComboBox.tsx:151]: Finally got around to testing this. Its not what I expected (forgot that blurring the combobox clears user input, so _less_ of a concern to filter the list based on what the user has already typed)

But we should probably at least clear the input, especially if the selection is not a possible choice after the options change:

combobox

Comment (werdnanoslen) [reply src/components/forms/ComboBox/ComboBox.tsx:151]: I had the same thought, but left it off to preserve the simplicity of the example's focus even if impractical for actual use. Also, adding a `ref`-based clear on the radio change also prompts the options to change without needing the useEffect, but I figured that shouldn't be the ideal fix here. So I'll add the clear and keep the useEffect too.

Comment (werdnanoslen) [reply src/components/forms/ComboBox/ComboBox.tsx:151]: Feb-07-2024 16-45-22

**Review by shkeating — APPROVED** [review]
looks good to me!

**Comment by shkeating** [comment]
oop @brandonlenz i didn't realize this was still waiting on review from you 😳 i didn't know it would let me merge without code owners review

**Pull request merged**
